main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -335,6 +344,15 @@ object TypeReduction {
     }
   }
 
+  private def getField(clazz: Class[_], fieldName: String): Option[Field] = {
+    if (clazz.isArray && fieldName == "length") None // TODO this should return some of the length field
+    else try {
+      Some(clazz.getField(fieldName))
```

Comment:
What about field visibility?

Reply:
The method code doesn't have anything checking for that, but good point. follow-up PR?

Reply:
OK